Add support for FreeBSD

I manage the gitlab package for FreeBSD.
Gitlab started to use zoekt for an indexer.

While I try to create a package for FreeBSD, I get the error message:
```
Acquiring build lock
# github.com/sourcegraph/zoekt/index
vendor/github.com/sourcegraph/zoekt/index/merge.go:152:20: undefined: NewIndexFile
vendor/github.com/sourcegraph/zoekt/index/read.go:634:16: undefined: NewIndexFile
vendor/github.com/sourcegraph/zoekt/index/read.go:723:14: undefined: NewIndexFile
Releasing build lock

```

The problem is related to this line:
https://github.com/sourcegraph/zoekt/blob/main/index/indexfile.go#L15

Do you see a problem to add support for FreeBSD?

Thanks
Matthias

--- a/index/indexfile.go
+++ b/index/indexfile.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package index
 

